packages/payload/src/collections/operations/findByID.ts

```diff
@@ -170,8 +170,20 @@ export const findByIDOperation = async <
       throw new NotFound(t)
     }
 
+    // Query the document first
+    const docFromDB = await req.payload.db.findOne(findOneArgs)
+
+    // Return NotFound when access control denies
+    if (!docFromDB && !args.data && !overrideAccess && accessResult !== true) {
```

Comment:
Feels like we just need 1 check `!docFromDB`, it either exists or it does not. Then you can remove L188-L194.